test/Microsoft.Extensions.Internal.Test/ObjectMethodExecutorTest.cs

```diff
@@ -427,6 +511,17 @@ public ValueTask<string> ValueTaskOfReferenceType(string result)
             public void MethodWithMultipleParameters(int valueTypeParam, string referenceTypeParam)
             {
             }
+
+            public FSharpAsync<string> FSharpAsyncMethod(string parameter)
```

Comment:
Adding a test project would be great, it would help us see exactly what is being supported, and would also make it easier to consider corner cases for cancellation, exceptions and so on